### ISSUE TITLE
Get and Patch requests for PartnerOrg, Agreement, Intervention, GovInt need to use the same serializer

### ISSUE THREAD

**Issue by robertavram** [unicef/etools#429]

**Comment by robertavram** [comment]
@ilcsik actually see if you can just use the detail serializer to return the response after the patch request... for ex: the patch request uses CreateUpdate serializer to validate the data ... however, it should use the DetailSerializer after it's done... see if there is an easy way to do that

**Comment by ilcsik** [comment]
@robertavram I'm not sure what you're trying to achieve here. Validate with the CreateUpdate, but return the response with regular DetailSerializer if I get this right? What's the reason behind this?

To be able to return a serializer with data the whole validation etc cycle needs to be run via is_valid method so that the data is prepared in the .data attribute - that'd be an extra "empty" serialization flow. 

If it's a business requirement to get the same data after patching like you get after GET, we might redirect the call if its successful the the retrieve endpoint but that's an extra DB call.

**Comment by ilcsik** [comment]
Or maybe there is a way, with .initial_data, I just tried this:
```python
-        return Response(serializer.data)
-
+        return Response(AgreementRetrieveSerializer(data=serializer.data).initial_data)

```

Is this what you mean? If I make a PR will you check if this is OK?

**Comment by robertavram** [comment]
Does it work ok? It looks reasonable.. for example on the AgreementSerializer.. try a patch updating the authorized officers. @ilcsik

**Comment by ilcsik** [comment]
It works, does not change the PATCH flow - it only serializers after that, right before return. I think there could be a problem though, when the Retrieve serializer uses a field that is not present in CreateUpdate data, since then we won't be able to get that.

So I modified to use the instance instead of the serializer data:

```python
return Response(AgreementRetrieveSerializer(instance).data)
```

So in this way we have all the data needed, for eg. auth officers details like name etc. Seems fine.

**Comment by robertavram** [comment]
@ilcsik perfect

**Comment by ilcsik** [comment]
OK I'll submit a PR that has all 4 endpoints modified.

**Comment by ilcsik** [comment]
Resolved in #435. I havent modified GovInt view since it is already using the same serializer for GET and PATCH.

**Comment by caktus-philip** [comment]
Resolved by #435
